centreon-certified/clickhouse/clickhouse-metrics-apiv2.lua

```diff
@@ -0,0 +1,509 @@
+      .. tostring(self.sc_params.params.reverse_category_mapping[category]) .. " and element: "
+      .. tostring(self.sc_params.params.reverse_element_mapping[category][element])
+      .. ". If it is a not a misconfiguration, you should create a format file to handle this kind of element")
+  else
+    self.format_event[category][element]()
+  end
+
+  self.sc_logger:debug("[EventQueue:format_event]: event formatting is finished")
+end
+
+--------------------------------------------------------------------------------
+---- EventQueue:format_event_host method
+--------------------------------------------------------------------------------
+function EventQueue:format_event_host()
+  local event = self.sc_event.event
+  self.sc_logger:debug("[EventQueue:format_event_host]: call build_metric ")
+  self.sc_metrics:build_metric(self.format_metric[event.category][event.element])
+end
+
+--------------------------------------------------------------------------------
+---- EventQueue:format_event_service method
+--------------------------------------------------------------------------------
+function EventQueue:format_event_service()
+  self.sc_logger:debug("[EventQueue:format_event_service]: call build_metric ")
+  local event = self.sc_event.event
+  self.sc_metrics:build_metric(self.format_metric[event.category][event.element])
+end
+
+--------------------------------------------------------------------------------
+---- EventQueue:format_metric_host method
+-- @param metric {table} a single metric data
+--------------------------------------------------------------------------------
+function EventQueue:format_metric_host(metric)
+  self.sc_logger:debug("[EventQueue:format_metric_host]: call format_metric ")
+  local event = self.sc_event.event
+  metric.custom_id = event.host_id .. "-" .. metric.metric_name
+  self:format_metric_event(metric)
+end
+
+--------------------------------------------------------------------------------
+---- EventQueue:format_metric_service method
+-- @param metric {table} a single metric data
+--------------------------------------------------------------------------------
+function EventQueue:format_metric_service(metric)
+  self.sc_logger:debug("[EventQueue:format_metric_service]: call format_metric ")
+  local event = self.sc_event.event
+  metric.custom_id = event.host_id .. "-" .. event.service_id .. "-" .. metric.metric_name
+  self:format_metric_event(metric)
+end
+
+--------------------------------------------------------------------------------
+---- EventQueue:format_metric_service method
+-- @param metric {table} a single metric data
+-------------------------------------------------------------------------------
+function EventQueue:format_metric_event(metric)
+  self.sc_logger:debug("[EventQueue:format_metric]: start real format metric ")
+  local event = self.sc_event.event
+  local params = self.sc_params.params
+
+  -- self.sc_event.event.formated_event = {
```

Comment:
Is it a development residue?

Reply:
Foque, github stole my comment

Reply:
This was the previous structure.
Then I had to add the current structure which is using a hack in order to retrieve the metric id when needed.
I've left the original structure commented because one day we will be able to store data right from the stream connector. This will allow us to remove my hack and switch back to the standard way of coding the data structure